metagpt/context.py

```diff
@@ -85,7 +85,12 @@ def llm(self) -> BaseLLM:
         # if self._llm is None:
         self._llm = create_llm_instance(self.config.llm)
         if self._llm.cost_manager is None:
-            self._llm.cost_manager = self.cost_manager
+            if self.config.llm.api_type == LLMType.FIREWORKS:
+                self._llm.cost_manager = FireworksCostManager()
+            elif self.config.llm.api_type == LLMType.OPEN_LLM:
+                self._llm.cost_manager = TokenCostManager()
+            else:
+                self._llm.cost_manager = self.cost_manager
```

Comment:
what about the cost_manager inside `llm_with_cost_manager_from_llm_config`. Would it be a single func to init cost_manager